internal/tplfuncs/stencil.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
+
+// Description: This file contains the public API for templates
+// for stencil
+
+package tplfuncs
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/getoutreach/stencil/internal/functions"
+	"github.com/getoutreach/stencil/pkg/configuration"
+)
+
+// Stencil contains the global functions available to a template for
+// interacting with stencil.
+type Stencil struct {
+	s *functions.Stencil
+	m *configuration.ServiceManifest
+}
+
+// Arg returns the value of an argument in the service's
+// manifest.
+// Note: Only the top-level arguments are supported.
+//
+//   {{- stencil.Arg "name" }}
+func (s *Stencil) Arg(path string) interface{} {
```

Comment:
Just FYI -- no action needed.

In my non-outreach version of bootstrap, I have the concept of `prompt args` -- `{{ stencil.Arg "foo" "Do you want foo?" }}` which allows bootstrap to prompt the user to fill out `foo` if it is not present.  In my case, I store this in the lockfile though we could very well store this in the main service.yaml itself...

(In my version, the `prompt` calls will fail in CI if the arg is not present; it won't prompt in CI)

In retrospect, I wish I had some sort of a schema for modules that would make it easy for `stencil describe module-foo` to spit out the arguments and their individual documentation..  So, it might be worth considering if we can add a `stencil.DocumentArg(...)` where the documentation is persisted in the lock file and used in stencil help.

Reply:
We actually store the docs in the `manifest.yaml` that exists in the module repository, we can totally expose the information in `stencil describe` though. The idea was if something is "required" we'd prompt for a value before rendering the templates -- of course, that's not actually done yet 😓